src/components/legend.ts

```diff
@@ -1,4 +1,233 @@
 namespace Plottable.Components {
+  /**
+   * The Legend's column representation. Stores position information
+   * as well as data
+   */
+  class LegendColumn<T> {
+    constructor(private _width = 0, private _height = 0, private _data: T) {}
+
+    /**
+     * @return {T} data Returns any data stored in the column
+     */
+    public getData() {
+      return this._data;
+    }
+
+    /**
+     * @return {number} height The height of the column.
+     */
+    public getHeight() {
+      return this._height;
+    }
+
+    /**
+     * @return {number} width The width of the column.
+     */
+    public getWidth() {
+      return this._width;
+    }
+
+    /**
+     * @param {number} width Sets the width of the column
+     */
+    public setWidth(width: number) {
+      return this._width = width;
+    }
+  }
+
+  /**
+   * The Legend's row representations. Stores positioning information
+   * and column data.
+   */
+  class LegendRow {
```

Comment:
hmm good question -- rotated text should still be doable by the table (rotate the table :p ) I think even center-aligned legends are still a subset of a grid, just one with preference for changing margins such that everything is centered by row. Thoughts?